python/lsst/pipe/tasks/insertFakes.py

```diff
@@ -342,17 +350,109 @@ def run(self, fakeCat, image, wcs, photoCalib):
+            galImages = self.mkFakeGalsimGalaxies(fakeCat[galaxies], band, photoCalib, pixelScale, psf, image)
+
+            stars = (fakeCat[self.config.sourceType] == starCheckVal)
+            starImages = self.mkFakeStars(fakeCat[stars], band, photoCalib, psf, image)
+
+        else:
+            galImages, starImages = self.processImagesForInsertion(fakeCat, wcs, psf, photoCalib, band,
+                                                                   pixelScale)
+
+        image = self.addFakeSources(image, galImages, "galaxy")
         image = self.addFakeSources(image, starImages, "star")
         resultStruct = pipeBase.Struct(imageWithFakes=image)
 
         return resultStruct
 
+    def processImagesForInsertion(self, fakeCat, wcs, psf, photoCalib, band, pixelScale):
+        """Process images from files into the format needed for insertion.
+
+        Parameters
+        ----------
+        fakeCat : `pandas.core.frame.DataFrame`
+                    The catalog of fake sources to be input
+        wcs : `lsst.afw.geom.skyWcs.skyWcs.SkyWc`
+                    WCS to use to add fake sources
+        psf : `lsst.meas.algorithms.coaddPsf.coaddPsf.CoaddPsf` or
+              `lsst.meas.extensions.psfex.psfexPsf.PsfexPsf`
+                    The PSF information to use to make the PSF images
+        photoCalib : `lsst.afw.image.photoCalib.PhotoCalib`
+                    Photometric calibration to be used to calibrate the fake sources
+        band : `str`
+                    The filter band that the observation was taken in.
+        pixelScale : `float`
+                    The pixel scale of the image the sources are to be added to.
+
+        Returns
+        -------
+        galImages : `list`
+                    A list of tuples of `lsst.afw.image.exposure.exposure.ExposureF` and
+                    `lsst.geom.Point2D` of their locations.
+                    For sources labelled as galaxy.
+        starImages :
+                    A list of tuples of `lsst.afw.image.exposure.exposure.ExposureF` and
+                    `lsst.geom.Point2D` of their locations.
+                    For sources labelled as star.
+
+        Notes
+        -----
+        The input fakes catalog needs to contain the absolute path to the image in the
+        band that is being used to add images to. It also needs to have the R.A. and
+        declination of the fake source in radians and the sourceType of the object.
+        """
+        galImages = []
+        starImages = []
+
+        for (imFile, sourceType, mag, x, y) in zip(fakeCat[band + "imFilename"].array,
+                                                   fakeCat["sourceType"].array,
+                                                   fakeCat[self.config.magVar % band].array,
+                                                   fakeCat["x"].array, fakeCat["y"].array):
+
+            im = afwImage.ImageF.readFits(imFile)
+
+            xy = geom.Point2D(x, y)
```

Comment:
This code block including the try/except appears in slightly modified form three times now, so you may want to consider putting it into a separate function with a note explaining its purpose (since it's only covered in the mkFakeStars docstring).

Reply:
I think each is different enough to leave it, I'll improve the docs though.